MapXContentParserTests.testSimpleMap throws NPE after Jackson upgrade

The `MapXContentParserTests.testSimpleMap` started failing in CI after merging #53523. 

```
org.elasticsearch.common.xcontent.MapXContentParserTests > testSimpleMap FAILED
    java.lang.AssertionError: expected:<AQID> but was:<null>
        at __randomizedtesting.SeedInfo.seed([B5D2BC2027A0BE5E:D322E4A542F2F2C]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.common.xcontent.MapXContentParserTests.compareTokens(MapXContentParserTests.java:97)
        at org.elasticsearch.common.xcontent.MapXContentParserTests.testSimpleMap(MapXContentParserTests.java:35)
REPRODUCE WITH: ./gradlew ':libs:elasticsearch-x-content:test' --tests "org.elasticsearch.common.xcontent.MapXContentParserTests.testSimpleMap" -Dtests.seed=B5D2BC2027A0BE5E -Dtests.security.manager=true -Dtests.locale=sr-ME -Dtests.timezone=Africa/Nouakchott -Dcompiler.java=13
```

https://gradle-enterprise.elastic.co/s/ui5rafibu6aak/tests/f2bx2zoqli3xy-bwbqzhxxgk63i

--- a/libs/x-content/src/test/java/org/elasticsearch/common/xcontent/MapXContentParserTests.java
+++ b/libs/x-content/src/test/java/org/elasticsearch/common/xcontent/MapXContentParserTests.java
@@ -25,6 +25,7 @@ import org.elasticsearch.common.xcontent.support.MapXContentParser;
 import org.elasticsearch.test.ESTestCase;
 
 import java.io.IOException;
+import java.util.EnumSet;
 import java.util.Map;
 
 import static org.elasticsearch.common.xcontent.XContentParserTests.generateRandomObject;
@@ -73,7 +74,13 @@ public class MapXContentParserTests extends ESTestCase {
     }
 
     public void compareTokens(CheckedConsumer<XContentBuilder, IOException> consumer) throws IOException {
-        final XContentType xContentType = randomFrom(XContentType.values());
+        for (XContentType xContentType : EnumSet.allOf(XContentType.class)) {
+            logger.info("--> testing with xcontent type: {}", xContentType);
+            compareTokens(consumer, xContentType);
+        }
+    }
+
+    public void compareTokens(CheckedConsumer<XContentBuilder, IOException> consumer, XContentType xContentType) throws IOException {
         try (XContentBuilder builder = XContentBuilder.builder(xContentType.xContent())) {
             consumer.accept(builder);
             final Map<String, Object> map;
@@ -94,7 +101,13 @@ public class MapXContentParserTests extends ESTestCase {
                         assertEquals(token, mapToken);
                         assertEquals(parser.currentName(), mapParser.currentName());
                         if (token != null && (token.isValue() || token == XContentParser.Token.VALUE_NULL)) {
-                            assertEquals(parser.textOrNull(), mapParser.textOrNull());
+                            if (xContentType != XContentType.YAML || token != XContentParser.Token.VALUE_EMBEDDED_OBJECT) {
+                                // YAML struggles with converting byte arrays into text, because it
+                                // does weird base64 decoding to the values. We don't do this
+                                // weirdness in the MapXContentParser, so don't try to stringify it.
+                                // The .binaryValue() comparison below still works correctly though.
+                                assertEquals(parser.textOrNull(), mapParser.textOrNull());
+                            }
                             switch (token) {
                                 case VALUE_STRING:
                                     assertEquals(parser.text(), mapParser.text());

